moj-node/test/repositories/SearchRepository.spec.js

```diff
@@ -0,0 +1,288 @@
+const searchRepository = require('../../server/repositories/search');
+
+describe('searchRepository', () => {
+  describe('#find', () => {
+    describe('When called with a search query', () => {
+      it('should return the results', async () => {
+        const mockResults = [
+          {
+            _id: '1',
+            _source: {
+              title: ['First Item'],
+              summary: ['First item summary'],
+            },
+          },
+          {
+            _id: '2',
+            _source: {
+              title: ['Second Item'],
+              summary: ['Second item summary'],
+            },
+          },
+          {
+            _id: '3',
+            _source: {
+              title: ['Third Item'],
+              summary: ['Third item summary'],
+            },
+          },
+        ];
+
+        const client = createClient({
+          body: { hits: { hits: mockResults } },
+        });
+
+        const repository = searchRepository(client);
+
+        const response = await repository.find({
+          query: 'Test Query',
+          prison: 'HMP Development',
+        });
+
+        expect(client.post.callCount).to.equal(
+          1,
+          'The method should have performed a post using the client',
+        );
+
+        expect(response.length).to.equal(
+          mockResults.length,
+          'The method should have returned the same number of results',
+        );
+
+        expect(response[0]).to.eql(
+          {
+            title: 'First Item',
+            summary: 'First item summary',
+            url: '/content/1',
+          },
+          'The method should have transformed the results',
+        );
+      });
+
+      it('should filter by prison', async () => {
+        const client = createClient({
+          body: { hits: { hits: [] } },
+        });
+
+        const repository = searchRepository(client);
+
+        await repository.find({
+          query: 'Test Query',
+          prison: 'HMP Development',
+        });
+
+        expect(client.post.callCount).to.equal(
+          1,
+          'The method should have performed a post using the client',
+        );
+
+        const requestBody = client.post.lastCall.args[1];
+
+        expect(requestBody.query.bool.should[0].term.prison_name).to.equal(
+          'HMP Development',
+          'The query should filter by prison name',
+        );
+
+        expect(
```

Comment:
Don't do this... your coupling the test with the with the search provider.. just check that that the `stringified` request body contains the prison name

Reply:
Stringified the requests body and added assertions for the prison name in the body